File: lol/src/test/kotlin/com/your/lol/support/JsonFileConverter.java
package com.your.lol.support;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import org.jetbrains.annotations.NotNull;
import org.springframework.core.io.ClassPathResource;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;

public class JsonFileConverter {

    private static final Gson gson = new Gson();

    private JsonFileConverter() {
    }

    /**
     * 일반 제네릭의 경우 아래 메서드를 사용을 권장합니다.
     */
    public static <T> T fromJsonFile(String filePath, Class<T> type) {
        BufferedReader bufferedReader = read(filePath);
        return gson.fromJson(bufferedReader, type);
    }

    /**
     * 복합 타입을 가진 제네릭의 경우 (슈퍼 타입 토큰) 아래 메서드 사용을 권장합니다.
     */
    public static <T> T fromJsonFile(String filePath, TypeToken<T> typeToken) {
        BufferedReader bufferedReader = read(filePath);
        return gson.fromJson(bufferedReader, typeToken);
    }

    @NotNull
    private static BufferedReader read(String filePath) {
        try {
            File jsonFile = new ClassPathResource(filePath).getFile();
            BufferedReader bufferedReader = new BufferedReader(new FileReader(jsonFile));
            return bufferedReader;
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: ggtics-recruitement-lol-score/src/test/kotlin/com/your/lol/support/JsonFileConverter.kt
package com.your.lol.support

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import org.springframework.core.io.ClassPathResource
import java.io.BufferedReader
import java.io.File
import java.io.FileReader
import java.io.IOException

object JsonFileConverter {

    private val gson: Gson = Gson()

    fun <T> fromJsonFile(filePath: String, type: Class<T>): T {
        val bufferedReader = readJsonFile(filePath)
        return gson.fromJson(bufferedReader, type)
    }

    fun <T> fromJsonFile(filePath: String, typeToken: TypeToken<T>): T {
        val bufferedReader = readJsonFile(filePath)
        return gson.fromJson(bufferedReader, typeToken)
    }

    private fun readJsonFile(filePath: String): BufferedReader {
        try {
            val jsonFile: File = ClassPathResource(filePath).file
            return BufferedReader(FileReader(jsonFile))
        } catch (exception: IOException) {
            throw RuntimeException(exception)
        }
    }
}